lapce-data/src/hover.rs

```diff
@@ -152,41 +170,317 @@ impl Default for HoverData {
+    pub fn from_data(data: &LapceTabData) -> Self {
+        Self {
+            base_font_size: data.config.editor.font_size as f64,
+            link_color: data
+                .config
+                .get_color_unchecked(LapceTheme::EDITOR_LINK)
+                .clone(),
+        }
+    }
+}
+
 /// A hover entry
 /// This is separate from the lsp-types version to make using it more direct
 #[derive(Clone)]
 pub enum HoverItem {
-    // TODO: This could hold the data needed to render the markdown output
-    Markdown(String),
+    /// Rendered (druid) markdown text
+    Markdown(MarkdownText),
     PlainText(String),
 }
 impl HoverItem {
+    pub fn as_markdown_text(&self) -> MarkdownText {
+        match self {
+            HoverItem::Markdown(text) => text.clone(),
+            // TODO: We could store PlainText as an ArcStr/related type to make these clones
+            // significantly cheaper
+            HoverItem::PlainText(text) => MarkdownText::new(
+                RichText::new(ArcStr::from(text.as_str())),
+                Vec::new(),
+            ),
+        }
+    }
+
     pub fn as_str(&self) -> &str {
         match self {
-            HoverItem::Markdown(x) | HoverItem::PlainText(x) => x.as_str(),
+            HoverItem::Markdown(x) => x.as_str(),
+            HoverItem::PlainText(x) => x.as_str(),
         }
     }
-}
-impl From<MarkedString> for HoverItem {
-    fn from(text: MarkedString) -> Self {
+
+    fn from_marked_string(text: MarkedString, style: HoverTextStyle) -> Self {
         match text {
-            MarkedString::String(text) => HoverItem::Markdown(text),
+            MarkedString::String(text) => {
+                HoverItem::Markdown(parse_markdown(&text, style))
+            }
             // This is a short version of a code block
             MarkedString::LanguageString(code) => {
+                // TODO: We could simply construct the MarkdownText directly
                 // Simply construct the string as if it was written directly
-                HoverItem::Markdown(format!(
-                    "```{}\n{}\n```",
-                    code.language, code.value
+                HoverItem::Markdown(parse_markdown(
+                    &format!("```{}\n{}\n```", code.language, code.value),
+                    style,
                 ))
             }
         }
     }
-}
-impl From<MarkupContent> for HoverItem {
-    fn from(content: MarkupContent) -> Self {
+
+    fn from_markup_content(content: MarkupContent, style: HoverTextStyle) -> Self {
         match content.kind {
-            MarkupKind::Markdown => HoverItem::Markdown(content.value),
+            MarkupKind::Markdown => {
+                HoverItem::Markdown(parse_markdown(&content.value, style))
+            }
             MarkupKind::PlainText => HoverItem::PlainText(content.value),
         }
     }
 }
+
+const BLOCKQUOTE_COLOR: Color = Color::grey8(0x88);
+
+#[derive(Data, Clone)]
+pub struct MarkdownText {
```

Comment:
This struct isn't (or at least, it doesn't look) really markdown-specific. Maybe a name like `StyledText` would be a better choice long term (but I'm really bad at naming, so maybe not).

Reply:
I think it is currently specific to markdown, even if it could be used more generally.

Reply:
It's used for markdown specifically, but the implementation/general purpose (holding even richer text than `RichText`) isn't related to markdown, I think. Anyway, it's fine.